### DISCUSSION TITLE
Optionally unset a linkopt

### DISCUSSION THREAD

**Discussion by sluongng** [bazelbuild/bazel#20837] [Q&A]
My use case is that we are currently doing this

```
common --enable_platform_specific_config=true

...

common:macos --linkopt="-Wl,-no_warn_duplicate_libraries"
common:macos --host_linkopt="-Wl,-no_warn_duplicate_libraries"
```

to silent the Xcode 15 warning about duplicated -lm flags.

However, on MacOS, we could optionally enable `--config=remote` to build remotely on Linux RBE.
And our linker on Linux does not like to have those flags passed to it.

```
/usr/bin/ld.gold: fatal error: -pie and -static are incompatible
collect2: error: ld returned 1 exit status
compilepkg: error running subcommand /usr/bin/gcc: exit status 1
```

(ignore the irrelevant error message, it's mostly just `ld.gold` linker not recognizing the flag above)

So ideally, I am looking for a way to set the above flags by default but wipe them when `--config=remote` is enabled.
I am looking into [platform_mapping](https://bazel.build/concepts/platforms#platform-mappings) right now as a potential solution, but the API is deemed to be unstable and the feature will go away in the future.

A simple solution for this would be to be able to set a default `--linkopt` in `.bazelrc` for macOS builds, but to have the ability to unset it when needed (i.e. when RBE is manually enabled with `--config=remote`).

**Comment by sluongng** [comment]
Ended using platform_mapping to solve my problem https://github.com/buildbuddy-io/buildbuddy/pull/5680/files

Would love to hear suggestions for alternatives.

**Reply by keith** [reply]
note that for this specific problem we solve this in the toolchain now with a feature, so you could use `build:remote --features=-no_warn_duplicate_libraries`

**Reply by sluongng** [reply]
cc: @brentleyjones 

I think this is only available in apple_support cc toolchain, which we recently disabled by default? 🤔

**Reply by brentleyjones** [reply]
Correct. We aren't using the apple support toolchain.
